### PR TITLE
fix cleanfigure for textboxes and 2D axis

### LINKED ISSUE #637: `develop` fails on ACID test 69
`testGraphical(69)` latexmath2 fails on R2014a, because `cleanfigure` removes the complete figure content on 0505932.

R2014b is fine.

### PR DESCRIPTION
**Pull request by PeterPablo**
fixes #637

* Furthermore improves ACID test by removing latex warning
* `ACID(69) latexmath2` is removed from list of unreliable tests on octave and HG2 (HG1 shows inconsistency of `at`, see #552).

### CODE REVIEW

File: src/cleanfigure.m [hunk 1]
```diff
@@ -582,3 +586,9 @@ function movePointsCloser(meta, handle)
 
 end
 % =========================================================================
+function bool = isAxis3D(axisHandle)
```
Comment (PeterPablo) [inline src/cleanfigure.m]: This is duplicated from `matlab2tikz.m`.

Comment (egeerardyn) [inline src/cleanfigure.m]: Can't you move it to the `src/private` folder then?

Comment (PeterPablo) [inline src/cleanfigure.m]: Let me try. Will add another commit shortly.

**Comment by egeerardyn** [comment]
For R2014b and Octave, `ACID(69)` and all other reliable tests are fine.

For R2014a, I get a mismatch on `ACID(69)` only:
```latex
% This file was created by matlab2tikz.
%
\documentclass[tikz]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{pgfplots}
\usepackage{grffile}
\pgfplotsset{compat=newest}
\usetikzlibrary{plotmarks}
\usepgfplotslibrary{patchplots}
\usepackage{amsmath}

\newlength\figureHeight \setlength{\figureHeight}{6cm}
\newlength\figureWidth \setlength{\figureWidth}{10cm}
\begin{document}
\begin{tikzpicture}

\begin{axis}[%
width=0.951\figureWidth,
height=\figureHeight,
at={(0\figureWidth,0\figureHeight)},
scale only axis,
clip=false,
every outer x axis line/.append style={black},
every x tick label/.append style={font=\color{black}},
xmin=   0,
xmax=   1,
every outer y axis line/.append style={black},
every y tick label/.append style={font=\color{black}},
ymin=   0,
ymax=   1,
hide axis,
axis x line*=bottom,
axis y line*=left
]
\node[right, align=left, text=black]
at (0.458cm,7.573cm) {$L\{f(t)\}  \equiv  F(s) = \int_0^\infty\!\!{e^{-st}f(t)dt}$};
\node[right, align=left, text=black]
at (0.458cm,5.037cm) {$e = \sum_{k=0}^\infty {\frac{1}{k!}} $};
\node[right, align=left, text=black]
at (0.458cm,2.501cm) {$m \ddot y = -m g + C_D \cdot {\frac{1}{2}}\rho {\dot y}^2 \cdot A$};
\node[right, align=left, text=black]
at (0.458cm,-0.035cm) {$\int_{0}^{\infty} x^2 e^{-x^2} dx = \frac{\sqrt{\pi}}{4}$};
\end{axis}
\end{tikzpicture}%
\end{document}
```
I've also tried to make sure my `get(0,'Default')` is empty, but to no avail. It influences the output (i.e. hash `8d24b2d8db272eb4dea74e75a346d45e`), but still doesn't match yours.

**Comment by PeterPablo** [comment]
I will check tomorrow morning. I fear it again is an unreliable `width` or `at`, see #552. Let's see...

**Comment by PeterPablo** [comment]
Here is the diff:
```
--- .../matlab2tikz/test/data/converted/test69-converted.tex
+++ (clipboard)
@@ -34,13 +34,13 @@
 axis y line*=left
 ]
 \node[right, align=left, text=black]
-at (0.479cm,7.586cm) {$L\{f(t)\}  \equiv  F(s) = \int_0^\infty\!\!{e^{-st}f(t)dt}$};
+at (0.458cm,7.573cm) {$L\{f(t)\}  \equiv  F(s) = \int_0^\infty\!\!{e^{-st}f(t)dt}$};
 \node[right, align=left, text=black]
-at (0.479cm,5.048cm) {$e = \sum_{k=0}^\infty {\frac{1}{k!}} $};
+at (0.458cm,5.037cm) {$e = \sum_{k=0}^\infty {\frac{1}{k!}} $};
 \node[right, align=left, text=black]
-at (0.479cm,2.51cm) {$m \ddot y = -m g + C_D \cdot {\frac{1}{2}}\rho {\dot y}^2 \cdot A$};
+at (0.458cm,2.501cm) {$m \ddot y = -m g + C_D \cdot {\frac{1}{2}}\rho {\dot y}^2 \cdot A$};
 \node[right, align=left, text=black]
-at (0.479cm,-0.028cm) {$\int_{0}^{\infty} x^2 e^{-x^2} dx = \frac{\sqrt{\pi}}{4}$};
+at (0.458cm,-0.035cm) {$\int_{0}^{\infty} x^2 e^{-x^2} dx = \frac{\sqrt{\pi}}{4}$};
 \end{axis}
 \end{tikzpicture}%
 \end{document}
```

It's only the coordinates that misbehave. I don't see a straight solution so will mark that test as unreliable for HG1.

**Comment by PeterPablo** [comment]
I have the suspision that `get(0,'ScreenPixelsPerInch')` plays a part in this. If I change it from my systems default `90` to some other number, I get a very similar diff.

Here is the output of `get(0)` on my freshly started R2014a:
```
	CallbackObject = []
	CommandWindowSize = [137 20]
	CurrentFigure = []
	Diary = off
	DiaryFile = diary
	Echo = off
	FixedWidthFontName = Courier New
	Format = short
	FormatSpacing = loose
	Language = de_de
	MonitorPositions = [0 0 3200 1200]
	More = off
	PointerLocation = [2410 195]
	RecursionLimit = [500]
	ScreenDepth = [24]
	ScreenPixelsPerInch = [90]
	ScreenSize = [1 1 3200 1200]
	ShowHiddenHandles = off
	Units = pixels

	BeingDeleted = off
	ButtonDownFcn = 
	Children = []
	Clipping = on
	CreateFcn = 
	DeleteFcn = 
	BusyAction = queue
	HandleVisibility = on
	HitTest = on
	Interruptible = on
	Parent = []
	Selected = off
	SelectionHighlight = on
	Tag = 
	Type = root
	UIContextMenu = []
	UserData = []
	Visible = on
```

**Comment by egeerardyn** [comment]
:+1:

**Comment by PeterPablo** [comment]
Do you mind posting your `get(0)` of R2014a? (May be in #542 or #552.)

**Comment by PeterPablo** [comment]
Applying, `set(0,'ScreenPixelsPerInch',72.0567)`, as suggested by your configuration (https://github.com/matlab2tikz/matlab2tikz/issues/542#issuecomment-99396009) makes the output identical! :+1: 

This is a strong indication, that we should follow this approach in #542 to solve #552.

**Comment by PeterPablo** [comment]
@egeerardyn, please verify that `set(0,'ScreenPixelsPerInch',90)` makes `ACID(69)` pass on your R2014a.

**Comment by egeerardyn** [comment]
@PeterPablo I can confirm that with HEAD at ef735dc and setting `set(0,'ScreenPixelsPerInch',90)` passes at my side. So I concur with your conclusion.

**Pull request merged**